server/src/io/initialhandler.rs

```diff
@@ -280,14 +368,24 @@ fn handle_login_start(ih: &mut InitialHandler, packet: &LoginStart) -> Result<()
         );
         send_packet(ih, encryption_request);
 
+        ih.info = Some(JoinResult::with_username(packet.username.clone()));
+
         ih.stage = Stage::AwaitEncryptionResponse;
     } else {
-        // Finished - set info and join
-        ih.info = Some(JoinResult {
-            username: ih.username.clone().unwrap(),
-            uuid: Uuid::new_v4(),
-            props: vec![],
-        });
+        let username = packet.username.clone();
+
+        // Check if there is some info about the client available. This can be the case if the
+        // handshake is made by an IP forwarding proxy.
+        if ih.info.is_some() {
```

Comment:
I believe `if let Some(info) = ih.info.as_mut()` would be more idiomatic here.

Reply:
I agree. I have changed it in fe5e9eead5f3009ad708646ffb9c9cb9f37c9038.